en/debug/plotting_realtime_uorb_data.md

```diff
@@ -0,0 +1,118 @@
+# Plotting uORB Topic Data in Real Time using PlotJuggler
+
+This topic shows how you can graph the "live" values of [uORB topics](../msg_docs/index.md) (in real time) using [PlotJuggler](../log/flight_log_analysis.md#plotjuggler) and the _uXRCE-DDS Agent_.
+
+This technique uses PX4 [uXRCE-DDS](../middleware/uxrce_dds.md) middleware to export uORB topics as ROS2 topics, which can then be read and plotted by PlotJuggler as they change (PlotJuggler cannot directly read uORB topics, but the values of the corresponding ROS 2 topics are the same).
+
+The video below demonstrates this for a simulated vehicle — the approach works equally well on real hardware.
+
+<video src="../../assets/debug/realtime_debugging/realtime_debugging.mp4" width="720" controls></video>
+
+## Prerequisites
+
+Follow the [ROS 2 Installation & Setup](../ros2/user_guide.md#installation-setup) instructions in the _ROS2 user guide_ to install:
+
+- ROS 2
+- [Micro XRCE-DDS Agent](../ros2/user_guide.md#setup-micro-xrce-dds-agent-client)
+- [PX4/px4_msgs](https://github.com/PX4/px4_msgs): PX4/ROS2 shared message definitions.
+- PX4 source code and build the simulator.
+
+  ::: tip
+  If you're using real hardware instead of the simulator, you will only need PX4 source code if you need to change the set of topics that are published to ROS 2 (only a subset of uORB topics are published by default).
+  :::
+
+You will also need to install:
+
+- [PlotJuggler for ROS2](https://github.com/facontidavide/PlotJuggler)
+
+  ::: tip
+  Use the Debian packages (the snap files are not supported).
+  :::
+
+## Usage
+
+First we need to build a ROS 2 workspace that includes the `px4_msgs` that correspond to the PX4 build to be monitored, and then launch PlotJuggler from within that workspace.
+This allows ROS 2 and PlotJuggler to interpret the messages.
+If you're using unmodified PX4, the definitions from [PX4/px4_msgs](https://github.com/PX4/px4_msgs) can be used.
+
+::: info
+This is the same process as covered in [Build ROS 2 Workspace](../ros2/user_guide.md#build-ros-2-workspace) in _ROS 2 Installation & Setup_.
+:::
+
+Assuming your ROS 2 workspace is `~/ros2_ws/`, fetch and build the `px4_msgs` package in a terminal as shown:
+
+```sh
+cd ~/ros2_ws/src/
+git clone https://github.com/PX4/px4_msgs.git
+cd ..
+colcon build
+source install/setup.bash
```

Comment:
Is this correct? The build guide seems to indicate I should do this in `src`
```
source /opt/ros/humble/setup.bash
colcon build
```

Then the commands here: https://docs.px4.io/main/en/ros2/user_guide.html#running-the-example

Can you sanity check. The requirement is that this should work for a NEW workspace and fresh terminal after they have already installed ROS etc.

Reply:
yes, `source /opt/ros/humble/setup.bash` is for the ros environment, but you also need to source your workspace after building.
`source /opt/ros/humble/setup.bash` is not needed here as in the ros2 guide you're adding it to your `.bashrc`
but you could also add it here just in case.
I quickly tested it in a new install, and all dependencies are correct.
Only think I'm adding is to move to the PX4-Autopilot folder before `make px4_sitl gz_x500`

Reply:
Good idea. I did that.

Also split usage into two subsection sim/hardware. That makes it a little more digestable.